### PR TITLE
Window size limits

### PR DESCRIPTION
**Pull request by toasty1307**
### Added window size limits

* Added `MinimumSize` and `MaximumSize` to `NativeWindow` and `NativeWindowSettings`
* Affects Windowing.Desktop

### Testing status

* Tested both the properties in NativeWindowSettings when creating a window
* Tested changing the properties at runtime

### CODE REVIEW

**Comment by NogginBops** [comment]
Great PR!
We should document that if you pass `-1` (`GLFW_DONT_CARE`) as one of the dimensions to say there is no minimum or maximum.

**Comment by toasty1307** [comment]
Added the documentation!

File: src/OpenTK.Windowing.Desktop/NativeWindow.cs
Comment (NogginBops) [inline src/OpenTK.Windowing.Desktop/NativeWindow.cs:495]: I would say `aspect ratio lock` instead of just `aspect ratio` as I think it's somewhat unclear otherwise.

File: src/OpenTK.Windowing.Desktop/NativeWindow.cs
Comment (NogginBops) [inline src/OpenTK.Windowing.Desktop/NativeWindow.cs:542]: I think we want to say that this locks the window to this aspect ratio, but otherwise I think this is fine.

File: src/OpenTK.Windowing.Desktop/NativeWindowSettings.cs
Comment (NogginBops) [inline src/OpenTK.Windowing.Desktop/NativeWindowSettings.cs:179]: It's not only the initial aspect ratio, it's locked to this aspect ratio unless you change it later.

**Review by NogginBops — CHANGES_REQUESTED** [review]
I just have some small comments about the documentation then I think this PR is ready to merge!
Really appreciate the PR 🙂

**Review by utkumaden — COMMENTED** [review]
I'm going to leave a small note about the double call to `glfwSetWindowSizeLimits` on back to back call to `MinSize` and `MaxSize`. I would assume the functions is practically free. The alternative is having a method which would not look good.

Otherwise no problems I see.

**Comment by NogginBops** [comment]
Is this still WIP? @toasty1307

**Comment by toasty1307** [comment]
I forgot about this PR, sorry, no, not WIP anymore, everything should be done(?) @NogginBops

File: src/OpenTK.Windowing.Desktop/NativeWindowSettings.cs
Comment (NogginBops) [inline src/OpenTK.Windowing.Desktop/NativeWindowSettings.cs:167]: These should default to `null` instead of `GLFW.DontCare`.

File: src/OpenTK.Windowing.Desktop/NativeWindowSettings.cs
Comment (NogginBops) [inline src/OpenTK.Windowing.Desktop/NativeWindowSettings.cs:176]: This too.

File: src/OpenTK.Windowing.Desktop/NativeWindowSettings.cs
Comment (NogginBops) [inline src/OpenTK.Windowing.Desktop/NativeWindowSettings.cs:185]: And this one too.

**Review by NogginBops — CHANGES_REQUESTED** [review]

**Comment by NogginBops** [comment]
@toasty1307 I did find a little thing that we should change before merging. I'm happy to merge after those changes :slightly_smiling_face:

**Comment by NogginBops** [comment]
Great stuff!
Will merge this now.
Really appreciate the PR! :slightly_smiling_face:

**Pull request merged**
